### PR TITLE
[MRG] add drive warnings

### LINKED ISSUE #269: raise warning when no external drives added
> @cjayb could we perhaps raise a warning when running a simulation with no input drives? And telling users that they must add the drives manually or use add_drives_from_params=True. If they use add_drives_from_params=True, a big fat deprecation warning should be raised.

+1 on both counts. 

> perhaps we could not add a drive to net.external_drives if in the param file t0 of drive > tstop of simulation?

and

> The items in net.external_drives should map 1-to-1 with whatever is specified by a param file or the user via calls to net.add_... so that we don't end up silently loosing track of network drives, gids, etc.

I propose raising a `ValueError` for drives (from params or otherwise) where `t0 > T`. The solution is either to fix the param-file (setting weights to zero effectively "removes" the drive), or, preferably, use the new API.

### PR DESCRIPTION
**Pull request by orbekolo**
Closes #269

### CODE REVIEW

**Comment by ntolley** [comment]
@orbekolo this would be a good place to add a test for checking if the warning message is raised:
https://github.com/jonescompneurolab/hnn-core/blob/b630f931b1f52f8ab7c1c445ca585a6aeeffb545/hnn_core/tests/test_network.py#L477

This function checks if a warning is raised when there are no connections in the local network. You can use the same code to check if your warning for the drives is raised. `match` is just a criteria for the test to pass, as in some part of the warning message must contain that string.

After this function, you can use `net.add_connection()` to add any connection (and avoid the no connections warning). And then inside the test you can run `simulate_dipole()`. The drives warning should be raised since `net.clear_drives()` is called just a few lines up.

**Comment by jasmainak** [comment]
@orbekolo just checking in here. Do you need help? We're going to revamp the contributing guide ... any feedback would be most welcome!

**Comment by ntolley** [comment]
@orbekolo for some reason the changes to `whats_new.rst` didn't get updated here. You may need to try `git commit`-ing the file again and push again.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -21,12 +21,14 @@ Changelog
...
 
- - Add ability to target specific sections when adding drives or connections,
-   by `Nick Tolley`_ in :gh:`419`
+- Add ability to target specific sections when adding drives or connections,
+  by `Nick Tolley`_ in :gh:`419`
 
 - Runtime output messages now specify the trial with which each simulation time
   checkpoint belongs too, by `Ryan Thorpe`_ in :gh:`546`.
 
+- Add warning if network drives are not loaded, by `Orsolya Beatrix Kolozsvari`_ in :gh:`#516`
```
Comment (ntolley) [inline doc/whats_new.rst:30]:
```suggestion
- Add warning if network drives are not loaded, by `Orsolya Beatrix Kolozsvari`_ in :gh:`516`
```

**Comment by jasmainak** [comment]
@chenghuzi I let you merge if you're happy!

**Comment by chenghuzi** [comment]
> @chenghuzi I let you merge if you're happy!

LGTM!

**Comment by ntolley** [comment]
@orbekolo congrats on the first PR merged! :tada: :tada:

**Pull request merged**